flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/mailbox/MailboxProcessor.java

```diff
@@ -195,7 +195,12 @@ public void runMailboxLoop() throws Exception {
 	public void reportThrowable(Throwable throwable) {
 		sendControlMail(
 			() -> {
-				throw WrappingRuntimeException.wrapIfNecessary(throwable);
+				if (throwable instanceof Exception) {
+					throw (Exception) throwable;
+				}
+				else {
+					throw WrappingRuntimeException.wrapIfNecessary(throwable);
```

Comment:
Should we also skip wrapping of `Error`s? Previously, we didn't skip, so maybe another PR?.

Reply:
We have to wrap anything that's not an `Exception` (because of `RunnableWithException`), so we can't NOT wrap `Error`, otherwise code would not compile?

Reply:
I meant logic like this:
```
				if (throwable instanceof Exception) {
					throw (Exception) throwable;
				} else if (throwable instanceof Error) {
					throw (Error) throwable;
				} else {
					throw WrappingRuntimeException.wrapIfNecessary(throwable);
				}
```
My `javac` compiles it :)